resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceRequestHelperSuite.scala

```diff
@@ -50,7 +54,53 @@ class ResourceRequestHelperSuite extends SparkFunSuite with Matchers {
     sparkConf.set(DRIVER_CORES.key, "4")
     sparkConf.set(EXECUTOR_MEMORY.key, "4G")
     sparkConf.set(EXECUTOR_CORES.key, "2")
-    ResourceRequestHelper.validateResources(sparkConf)
+    validateResources(sparkConf)
+  }
+
+  test("get yarn resources from configs") {
+    val sparkConf = new SparkConf()
+    val resources = Map(YARN_GPU_RESOURCE_CONFIG -> "2G",
+      YARN_FPGA_RESOURCE_CONFIG -> "3G", "custom" -> "4")
+    resources.foreach { case (name, value) =>
+      sparkConf.set(s"${YARN_EXECUTOR_RESOURCE_TYPES_PREFIX}${name}.${AMOUNT}", value)
+      sparkConf.set(s"${YARN_DRIVER_RESOURCE_TYPES_PREFIX}${name}.${AMOUNT}", value)
+      sparkConf.set(s"${YARN_AM_RESOURCE_TYPES_PREFIX}${name}.${AMOUNT}", value)
+    }
+    var parsedResources = getYarnResourcesAndAmounts(sparkConf, YARN_EXECUTOR_RESOURCE_TYPES_PREFIX)
+    assert(resources.size === 3)
```

Comment:
I find that these tests generally fail with a very unhelpful message.

Since `getYarnResourcesAndAmounts` returns a map, isn't this + the check below basically the same as:

    assert(parsedResources === resources)

Reply:
yes, updated